lib/req_snowflake/snowflake_login.ex

```diff
@@ -0,0 +1,99 @@
+defmodule ReqSnowflake.SnowflakeLogin do
+  alias ReqSnowflake.Snowflake
+
+  def get_snowflake_login_token(options) when is_map(options) do
+    Keyword.new(options)
+    |> get_snowflake_login_token()
+  end
+
+  def get_snowflake_login_token(options \\ []) when is_list(options) do
+    data = build_snowflake_login(options)
+    url = snowflake_url(options)
+
+    Req.post!(url, json: data)
+    |> decode_response()
+  end
+
+  defp snowflake_url(options) when is_list(options) do
+    account_name = options[:account_name]
+    region = options[:region]
+
+    base_url = Snowflake.snowflake_url(account_name, region)
+
+    query_params =
+      %{
+        databaseName: Keyword.get(options, :database, nil),
+        schemaName: Keyword.get(options, :schema, nil),
+        roleName: Keyword.get(options, :role, nil),
+        warehouse: Keyword.get(options, :warehouse, nil)
+      }
+      |> Enum.filter(fn {_, v} -> v != nil end)
+      |> URI.encode_query()
+
+    "#{base_url}/session/v1/login-request"
+    |> URI.parse()
+    |> Map.put(:query, query_params)
+    |> URI.to_string()
+  end
+
+  defp decode_response(%Req.Response{status: 200, body: %{"data" => %{"token" => token}}}),
+    do: token
+
+  defp decode_response(%Req.Response{body: %{"message" => message}}),
+    do: RuntimeError.exception(message)
```

Comment:
Do you mean without the RuntimeError?
ie:
```
  defp decode_response(%Req.Response{body: %{"message" => message}}),
    do: message
```
or returning the actual response?
```
  defp decode_response(%Req.Response{body: %{"message" => _message}} = response),
    do: RuntimeError.exception(response)
```

Reply:
I like it, the main reason I did it the other way was because we can expose the error message Snowflake gives back to us, but for a client it's fine. That should be the responsibility the user or snowflake_elixir (db_connection) or whatever.